### PR TITLE
Add support for v2 ManageEvent api call

### PR DESCRIPTION
**Pull request by luqasn**
I took the liberty of adding the spike by @tmccleve from #83 to my fork in order to support creation of v2 API events. The code is almost identically to the v1 code that is already in the package apart from the JSON structures and the endpoint url.

### CODE REVIEW

**Comment by luqasn** [comment]
Can we get this merged?

**Comment by v-antech** [comment]
I would also like see if this can be merged. Looks fine to me. Also why is the library stating that it is supporting the V2 API, when in fact it is not? This seems a very essential feature to me.

**Comment by dan-force** [comment]
My team could really benefit from this change being merged. How can we help to make this happen? Who is in charge, btw? Can you help us out @istvandn or @luqasn?

**Comment by luqasn** [comment]
I have no idea who is in charge, but this library sadly seems to be pretty much abandoned by pagerduty. I think a new project should be created that is based on the swagger api definition for the v2 api. This would make keeping up to date much easier.

File: event_v2.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Event includes the incident/alert details
+type V2Event struct {
+	RoutingKey string        `json:"routing_key"`
+	Action     string        `json:"event_action"`
```
Comment (nuptse) [inline event_v2.go]: Please rename this to `EventAction` for consistency with the JSON field name

File: event_v2.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	Severity  string      `json:"severity"`
+	Timestamp string      `json:"timestamp,omitempty"`
+	Component string      `json:"component,omitempty"`
+	Group     string      `json:"group,omitempty"`
+	Class     string      `json:"class,omitempty"`
+	Details   interface{} `json:"custom_details,omitempty"`
+}
+
+// Response is the json response body for an event
+type V2EventResponse struct {
```
Comment (nuptse) [inline event_v2.go]: Based on our [current API documentation for the Events API v2](https://v2.developer.pagerduty.com/docs/events-api-v2), the response format is different than what is expected here.

File: event_v2.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		return nil, err
+	}
+	req, _ := http.NewRequest("POST", v2eventEndPoint, bytes.NewBuffer(data))
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusAccepted {
```
Comment (nuptse) [inline event_v2.go]: The API also returns validation errors back, it'll be great to have those displayed back to the user.

**Comment by mattstratton** [comment]
I haven't dug into this PR enough yet, but is it backwards compatible with v1? If not, we will want to be very clear with this release that it contains breaking changes. 

I am intending to add binary release options via goreleaser shortly; we should wait to merge this until we have the major versions cut, which I hope to do within the next couple days.

**Comment by jamiehannaford** [comment]
@mattstratton Hey! Would it be possible to add this soon? Since the struct names contain `v2` I think it's backwards compatible.

**Pull request merged**